radixdlt-java/src/main/java/com/radixdlt/client/application/translate/tokens/CreateTokenAction.java

```
@@ -16,23 +15,23 @@
 	private final String name;
 	private final String iso;
 	private final String description;
-	private final UInt256 initialSupply;
-	private final UInt256 granularity;
+	private final BigDecimal initialSupply;
+	private final BigDecimal granularity;
 	private final RadixAddress address;
 	private final TokenSupplyType tokenSupplyType;
 
-	public CreateTokenAction(
+	private CreateTokenAction(
 		RadixAddress address,
 		String name,
 		String iso,
 		String description,
-		UInt256 initialSupply,
-		UInt256 granularity,
+		BigDecimal initialSupply,
+		BigDecimal granularity,
 		TokenSupplyType tokenSupplyType
 	) {
 		// Redundant null check added for completeness
 		Objects.requireNonNull(initialSupply);
-		if (tokenSupplyType.equals(TokenSupplyType.FIXED) && initialSupply.isZero()) {
+		if (tokenSupplyType.equals(TokenSupplyType.FIXED) && initialSupply.compareTo(BigDecimal.ZERO) <= 0) {
```

initialSupply < 0 should be an error, rather than silently accepted here.